### ISSUE TITLE
Materialize failing on valid schema ref

### ISSUE THREAD

**Issue by jacksmith15** [jacksmith15/json-ref-dict#14]
Odd behaviour where the absolute URI reference isn't updated wen propagating through in this case:
```python
from json_ref_dict import materialize, RefDict

ref_dict = RefDict("http://json.schemastore.org/swagger-2.0")

ref_dict["definitions"]["minLength"]
# {'allOf': [{'$ref': '#/definitions/positiveInteger'}, {'default': 0}]}

ref_dict["definitions"]["minLength"]["allOf"][0]
# {'type': 'integer', 'minimum': 0}

dictionary = materialize(ref_dict)
# JsonPointerException: member 'positiveInteger' not found in {'info': {'type': 'object', 'description': 'General information about the API.', 'required': ['version', 'title'], 'additionalProperties': False, 'patternProperties': {'^x-': {'$ref': '#/definitions/vendorExtension'}}...
```

But interestingly there is no problem with
```python
materialize(ref_dict, include_keys=("definitions", "minLength", "allOf", "default", "type"))
```

This doesn't occur everywhere so there must be something specific about this case, probably related to multiple references to the same point

**Comment by KissPeter** [comment]
I have the same issue, a workaround would be good until it is fixed

**Comment by jacksmith15** [comment]
Fixed in [0.7.1](https://pypi.org/project/json-ref-dict/0.7.1/), thanks to @evahala
